### PR TITLE
Getting security exception due to access denied 'java.lang.RuntimePermission' 'accessDeclaredMembers' when trying to get snapshot with S3 IRSA

### PR DESCRIPTION
**Pull request by reta**
Signed-off-by: Andriy Redko <andriy.redko@aiven.io>

### Description
Configuring to use IRSA via repository-s3 plugin for S3 access with identity file may end up with `SecurityException` due to access denied "java.lang.RuntimePermission" "accessDeclaredMembers" when trying to get snapshots.

### Issues Resolved
Closes https://github.com/opensearch-project/OpenSearch/issues/4269

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Service.java
Comment (dreamer-89) [inline plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Service.java:308]: Thanks @reta for the quick fix 👍 

For my understanding, I see the `getCredentials` on `PrivilegedSTSAssumeRoleSessionCredentialsProvider` is already wrapped under SocketAccess and so is `securityTokenService`, how does wrapping the `STSAssumeRoleSessionCredentialsProvider` builder fixes the issue ?

```
        public AWSCredentials getCredentials() {
            return SocketAccess.doPrivileged(credentials::getCredentials);
        }
```

nit: Do we need a unit test to verify the behavior ? I think we don't have tests verifying IRSA creds ? We can take this in separate PR to unblock this. 
https://github.com/opensearch-project/OpenSearch/blob/main/plugins/repository-s3/build.gradle#L113

Comment (reta) [reply plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Service.java:308]: @dreamer-89 that's the `STSAssumeRoleWithWebIdentitySessionCredentialsProvider.Builder::build()` and `STSAssumeRoleSessionCredentialsProvider.Builder::build()` which were not called in privileged context, the call to `getCredentials()` happens later on. 

We have covered IRSA with test cases and I have not seen this issue and third party integration tests, I think the flow with identity file is vulnerable to security checks, I will try to reproduce it in tests, it is quite very difficult.

Comment (dreamer-89) [reply plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3Service.java:308]: Thanks @reta for the clarification. I am good with the changes but would wait for one more approval.

**Review by dreamer-89 — APPROVED** [review]

**Review by kotwanikunal — APPROVED** [review]

**Comment by dreamer-89** [comment]
Thanks @kotwanikunal for the review. 
Resolved `Changelog.md` conflict manually which re-fired the different workflows

**Comment by dreamer-89** [comment]
@reta : The backports failed, are you planning to raise manual backports ?

**Comment by dblock** [comment]
@dreamer-89 I'm sure @reta would appreciate it if you beat him to it ;)

**Comment by reta** [comment]
> @reta : The backports failed, are you planning to raise manual backports ?

@dreamer-89 yes, on it, thanks!

**Comment by dreamer-89** [comment]
> @dreamer-89 I'm sure @reta would appreciate it if you beat him to it ;)

Sorry was bit late on it. Looks like @reta already completed it.

**Pull request merged**
